auth/presentation/src/main/java/com/sergius/auth/presentation/login/LoginViewModel.kt

```diff
@@ -22,38 +26,44 @@ class LoginViewModel(
     private val eventChannel = Channel<LoginEvent>()
     val events = eventChannel.receiveAsFlow()
 
-    init {
-        combine(
-            snapshotFlow { state.email.text },
-            snapshotFlow { state.password.text }
-        ) { email, password ->
-            val isEmailValid = userDataValidator.isValidEmail(email = email.toString())
-            state = state.copy(
-                isEmailValid = isEmailValid,
-                canLogin = isEmailValid && password.isNotEmpty()
-            )
-
-        }.launchIn(viewModelScope)
-    }
+    private val _isComposing = MutableStateFlow(false)
+    val isComposing = _isComposing
+        .onStart {
+            combine(
+                snapshotFlow { state.emailState.email.text },
+                snapshotFlow { state.passwordState.password.text }
```

Comment:
To avoid creating multiple snapshotFlows, you can leverage MutableStateFlow for managing your state. Here’s how:
```kotlin
private var _state = MutableStateFlow<LoginState>()
val state = _state
    .onStart {
        // Your logic here...
    }
    .stateIn(
        scope = viewModelScope,
        started = SharingStarted.WhileSubscribed(5_000L),
        initialValue = ChatDetailState()
    )
```
Then, to update values like email, stateState, or canLogin, you can simply use:
`_state.update { it.copy(/* your updated fields */) }`